tfe_hardware/qcodes_instrument_drivers/Current_source/AD5760.py

```diff
@@ -0,0 +1,138 @@
+        
+    def on(self) -> None:
+        self.write('OUTPUT 1')
+        self.measure._output = True
+        
+    def off(self) -> None:
+        self.write('OUTPUT 0')
+        self.measure._output = False
+
+    def state(self):
+        state = self.ask()
+        if state is not None:
+            return bool(1)
+        else:
+            return bool(0)
+    
+    def connect_board(self, host_name: str, path: str, syspath: str):
+        
+        sys.path.append(rf'{syspath}')
+        
+        # noinspection SpellCheckingInspection
+        
+        # noinspection SpellCheckingInspection
+        clr.AddReference('AnalogDevices.Csa.Remoting.Clients')
+        clr.AddReference('AnalogDevices.Csa.Remoting.Contracts')
+        
+        # noinspection PyUnresolvedReferences,SpellCheckingInspection
+        from AnalogDevices.Csa.Remoting.Clients import ClientManager  #noqa
+        manager = ClientManager.Create(-1)
+        self.client = manager.CreateRequestClient(f"localhost:{host_name}")
+        self.client.ContextPath = path
+        self.client.WriteRegister("2", "786")
+        
+    def _get_set_volt(self,
+                        output_level: Optional[float] = None
+                        ) -> Optional[float]:
+        """
+        Get or set the output level.
+
+        Args:
+            mode: "CURR" or "VOLT"
+            output_level: If missing, we assume that we are getting the
+                current level. Else we are setting it
+        """
+        # self._assert_mode(mode)
+        if output_level is not None:
+            self.write(output_level)
+            return None
+        return self.ask()
+
+        
+    def ramp_voltage(self, ramp_to: float, step: float, delay: float) -> None:
+        """
+        Ramp the voltage from the current level to the specified output.
+
+        Args:
+            ramp_to: The ramp target in Volt
+            step: The ramp steps in Volt
+            delay: The time between finishing one step and
+                starting another in seconds.
+        """
+        # self._assert_mode("VOLT")
+        self.ramp_trial(ramp_to, step, delay)
+
+
+    def ramp_trial(self, ramp_to: float, step: float, delay: float):
+        
+        v_in = self.voltage()
+        v_f = v_in
+        if ramp_to > v_in:
+            num_steps = (ramp_to - v_in)/step +1
+            for i in range(int(num_steps+1)):
+                v_f += step
+                self.voltage(v_f)
+                time.sleep(delay)
+        elif ramp_to < v_in:
+            num_steps = - (ramp_to - v_in)/step 
+            for i in range(int(num_steps+1)):
+                v_f -= step
+                self.voltage(v_f)
+                time.sleep(delay) 
+
+    def _get_volt(self):
+        return self.ask()
+
+    def _set_output(self, output_level) -> None:
+        self.write()
+        
+    def ask(self):
+        return (float(int(self.client.ReadRegister("1"), 16))-524288)*20/1048576
```

Comment:
why doesn't ask accept a parameter indicating what are you asking? does this instrument only have a single thing you are reading? For example [here](https://qcodes.github.io/Qcodes/drivers_api/Yokogawa.html#qcodes.instrument_drivers.yokogawa.YokogawaGS200.ask) ask accepts the string that is going to ask the instrument

Reply:
The instrument just has a single thing that we are reading....its an elementary setup with nothing else that can be asked

Reply:
you will never need to ask for register 2?